File: lia-agent-system/app/shared/seniority_levels.py
"""Niveis de senioridade — FONTE UNICA canonica (back + front).

Antes a lista de niveis vivia hardcoded no frontend
(compensation-policies-types.ts) e como strings soltas espalhadas (PRV, verba,
vaga, membro de depto). Aqui ela e a fonte unica: SalaryBand.level,
CompensationComponent.seniority_levels[] e a UI de bandas/escopo referenciam
estes ids. O espelho TS fica em plataforma-lia/src/lib/compensation/seniority-levels.ts
e DEVE permanecer identico (sensor: scripts/check_seniority_levels_sync.py).

Os ids normalizam corretamente via vacancy_vocab.to_match_seniority_key, entao o
matcher de elegibilidade continua tolerante a EN/PT/acentos.
"""
from __future__ import annotations

# Ordem = ordem de exibicao (junior -> c-level)
CANONICAL_SENIORITY_LEVELS: list[dict[str, str]] = [
    {"id": "junior", "label": "Júnior"},
    {"id": "pleno", "label": "Pleno"},
    {"id": "senior", "label": "Sênior"},
    {"id": "staff", "label": "Staff"},
    {"id": "principal", "label": "Principal"},
    {"id": "manager", "label": "Gerente"},
    {"id": "director", "label": "Diretor"},
    {"id": "c-level", "label": "C-Level"},
]

SENIORITY_LEVEL_IDS: tuple[str, ...] = tuple(l["id"] for l in CANONICAL_SENIORITY_LEVELS)

_LABEL_BY_ID = {l["id"]: l["label"] for l in CANONICAL_SENIORITY_LEVELS}

# Default de display order por id (para SalaryBand.order ao seedar)
_ORDER_BY_ID = {l["id"]: i for i, l in enumerate(CANONICAL_SENIORITY_LEVELS)}


def is_valid_level(level_id: str | None) -> bool:
    return bool(level_id) and level_id in SENIORITY_LEVEL_IDS


def label_for(level_id: str | None) -> str:
    if not level_id:
        return ""
    return _LABEL_BY_ID.get(level_id, level_id)


def order_for(level_id: str | None) -> int:
    if not level_id:
        return 999
    return _ORDER_BY_ID.get(level_id, 999)

File: plataforma-lia/src/lib/compensation/seniority-levels.ts
/**
 * Niveis de senioridade — ESPELHO TS da fonte unica canonica.
 *
 * DEVE permanecer identico a lia-agent-system/app/shared/seniority_levels.py
 * (mesmos ids, mesma ordem). Sensor: scripts/check_seniority_levels_sync.py.
 *
 * Antes a lista vivia hardcoded em compensation-policies-types.ts (SENIORITY_LEVEL_OPTIONS)
 * e em benefits-types.ts (SENIORITY_OPTIONS), divergentes. Esta e a fonte unica:
 * faixas salariais (SalaryBand.level), escopo de verba (seniority_levels[]) e a UI
 * de bandas referenciam estes ids.
 */
export interface SeniorityLevel {
  id: string
  label: string
}

export const CANONICAL_SENIORITY_LEVELS: readonly SeniorityLevel[] = [
  { id: "junior", label: "Júnior" },
  { id: "pleno", label: "Pleno" },
  { id: "senior", label: "Sênior" },
  { id: "staff", label: "Staff" },
  { id: "principal", label: "Principal" },
  { id: "coordinator", label: "Coordenador" },
  { id: "manager", label: "Gerente" },
  { id: "director", label: "Diretor" },
  { id: "c-level", label: "C-Level" },
] as const

export const SENIORITY_LEVEL_IDS: readonly string[] = CANONICAL_SENIORITY_LEVELS.map((l) => l.id)

/** Opcoes p/ ChipMultiSelect de escopo (com curinga "all" = aplica a todos). */
export const SENIORITY_SCOPE_OPTIONS: readonly SeniorityLevel[] = [
  { id: "all", label: "Todos" },
  ...CANONICAL_SENIORITY_LEVELS,
]

export function seniorityLabel(id: string): string {
  return CANONICAL_SENIORITY_LEVELS.find((l) => l.id === id)?.label ?? id
}

/** Tipos de contrato canonicos (escopo de verba/beneficio). */
export const CONTRACT_TYPE_OPTIONS: readonly SeniorityLevel[] = [
  { id: "clt", label: "CLT" },
  { id: "pj", label: "PJ" },
  { id: "estagio", label: "Estágio" },
  { id: "temporario", label: "Temporário" },
  { id: "freelancer", label: "Freelancer" },
  { id: "trainee", label: "Trainee" },
] as const
